src/mctpd.c

```diff
@@ -2893,6 +2893,49 @@ static int bus_link_get_prop(sd_bus *bus,
 	return rc;
 }
 
+static int notify_discovery(ctx *ctx, int ifindex)
+{
+	int rc = 0;
+	dest_phys desti = { 0 }, *dest = &desti;
+	struct mctp_ctrl_cmd_discovery_notify req = { 0 };
+	struct mctp_ctrl_resp_discovery_notify *resp;
+	uint8_t *buf;
+	size_t buf_size;
+	struct sockaddr_mctp_ext resp_addr;
+
+	dest->ifindex = ifindex;
+	assert(mctp_nl_ifaddr_byindex(ctx->nl, dest->ifindex,
```

Comment:
this will cause bugs with `NDEBUG`. Please split the assert from the call.

Reply:
Since we allowed for `0` address length, this assertion should not be necessary anymore.